frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -73,6 +75,12 @@ export default function SurveyEdit(): JSX.Element {
         setSelectedQuestion(newIndex)
     }
 
+    useEffect(() => {
+        if (survey.events && survey.events.length > 0) {
+            setSurveyValue('conditions', { url: '' })
+        }
+    }, [survey.events])
```

Comment:
I'm going to be annoying and say that we generally [discourage the use of useEffect()](https://posthog.com/handbook/engineering/conventions/frontend-coding), in our component code and that this should be a kea logic in `surveyLogic` instead (I'm mostly saying this because I had to spend a day figuring out how to do this for my HTML rendering work).

What you can do instead is create a set of kea logics here that are triggered inside the `onChange` down below that changes the underlying value of the `events` property of `survey`, and use those instead of relying on `useEffect()`.  It's a bit late and I'm a bit brain-fried so I can't come up with a cogent example for your use case, but here's how i accomplish a similar task with the HTML rendering code: https://github.com/PostHog/posthog/pull/22533/files#diff-6a4fe132d23700a91b23158c89998c1b7ebd180305f27f78bc0bb535a556154aR70-R101

Reply:
fixed, we don't need any of this by moving events into the conditions JSON blob